Coffe/Features/DrinkList/DrinkListViewController.swift

```diff
@@ -51,35 +141,37 @@ final class DrinkListViewController: UIViewController {
     // MARK: - Data Loading
     private func loadData() {
         Task {
-            await viewModel.fetchDrinks()
-            tableView.reloadData()
-            print("\(viewModel.filterCategories.count) count viewcontroller")
+            store.send(.loadDrinks)
         }
     }
 }
 // MARK: - UITableViewDataSource
 extension DrinkListViewController: UITableViewDataSource {
     func numberOfSections(in tableView: UITableView) -> Int {
-        return viewModel.filterCategories.keys.count
+        return store.filterCategories.keys.count
 
     }
     func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
-        return viewModel.filterCategories.keys.sorted()[section]
+        return store.filterCategories.keys.sorted()[section]
     }
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        let key = viewModel.filterCategories.keys.sorted()[section]
-        return viewModel.filterCategories[key]?.count ?? 0
+        let key = store.filterCategories.keys.sorted()[section]
+        return store.filterCategories[key]?.count ?? 0
     }
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: "DrinkTableViewCell", for: indexPath)
-        let key = viewModel.filterCategories.keys.sorted()[indexPath.section]
-        if let drink = viewModel.filterCategories[key]?[indexPath.row] {
+        let key = store.filterCategories.keys.sorted()[indexPath.section]
+        observations[indexPath]?.cancel()
+        observations[indexPath] = observe { [weak self] in
```

Comment:
@LukasSkrivanek Actually everything works even without it, but keeping it here if you would like to improve the logic